### PR TITLE
Fixed update of the shop name in the table ps_shop when not in multishop enabled

### LINKED ISSUE #23392: The shop name is the table ps_shop is not updated
#### Describe the bug

If your shop is not multishop :
When you change the shop name in the tab, it doesn't update the field name in the ps_shop table.
And that name is used in some scripts of the Core, for exemple in the e-mails (maybe in previous versions)

#### Expected behavior

The name sould update in the configuration but also in the table ps_shop

#### Steps to Reproduce

1. Go to "shop parameters" > "Contact" > tab "Stores".
2. Update the shop name
3. In the database go see in table ps_shop, the name is not updated

#### Additional information

* PrestaShop version: 1.7.7.1 & 8.0.4

### PR DESCRIPTION
**Pull request by dreammeup**
| Questions         | Answers
| ----------------- | -------------------------------------------------------
| Branch?           | develop
| Description?      | The shop name in the table ps_shop is not updating, and can't be updated when not in multishop enabled. The correction proposed : if we are not in multishop, we update the table with the value sended for PS_SHOP_NAME. This name is used in mails, maybe elsewhere.
| Type?             | bug fix
| Category?         | BO
| Deprecations?     | no
| How to test?      | Go in Admin tab "Stores", and edit the name of the shop, it won't change the name in "ps_shop" without this patch
| Fixed ticket? | Fixes #23392
| Possible impacts? | no

---
This change is [Reviewable](https://reviewable.io/reviews/prestashop/prestashop/23377)

### CODE REVIEW

**Comment by Progi1984** [comment]
@dreammeup Could you run `./vendor/bin/php-cs-fixer fix` on your local for fixing PHPCSFixer Github Action ?

**Review by matthieu-rolland — COMMENTED** [review]
Thank you @dreammeup for your contribution  👍 

Could you create an issue and reference it in your PR description ? Or reference the issue if you are treating an issue, this is mandatory in every PR ;)

(So that our product team can validate the existence of the bug and the wanted behavior, it will also help the QA team.)

**Comment by dreammeup** [comment]
Hello, the ticket was created and reviewed, I edited the pull request with the issue ticket number.

File: controllers/admin/AdminStoresController.php
Comment (PierreRambaud) [inline controllers/admin/AdminStoresController.php]:
```suggestion
            Db::getInstance()->execute(
```

File: controllers/admin/AdminStoresController.php
Comment (PierreRambaud) [inline controllers/admin/AdminStoresController.php]:
```suggestion
                'UPDATE `' . _DB_PREFIX_ . 'shop` ' .
                'SET name = "' . pSQL($value) . '" ' .
                'WHERE id_shop = ' . (int) $this->context->shop->id
```

**Review by sowbiba — DISMISSED** [review]

File: controllers/admin/AdminStoresController.php
Comment (matthieu-rolland) [inline controllers/admin/AdminStoresController.php]: It would be nice to have the shop ID as an optional parameter and use the context shop id if it's not specified. 👍

**Review by matthieu-rolland — COMMENTED** [review]
one small feedback

File: controllers/admin/AdminStoresController.php
Comment (PierreRambaud) [inline controllers/admin/AdminStoresController.php]:
```suggestion
    /**
     * @var string $name
     */
    public function updateOptionPsShopName(string $name): void
```

Could you improve the docstring to explained what this method do? :)

File: controllers/admin/AdminStoresController.php
Comment (PierreRambaud) [inline controllers/admin/AdminStoresController.php]:
```suggestion
                'SET name = "' . pSQL($name) . '" ' .
```

Related to my previous comment

File: controllers/admin/AdminStoresController.php
Comment (PierreRambaud) [inline controllers/admin/AdminStoresController.php]:
```suggestion
        if (!Shop::isFeatureActive() && !$this->errors && $name) {
```
Related to my previous comment

**Review by PierreRambaud — CHANGES_REQUESTED** [review]
Hey, thanks for your contribution, could you please improve docstring and use latest PHP7 feature (hint typing) please?

**Review by matthieu-rolland — COMMENTED** [review]
@dreammeup 

I put back the waiting for author label, don't hesitate to ask if there's a feedback that you don't quite understand 👍

**Review by PierreRambaud — APPROVED** [review]
Thanks for your reactivity!

**Review by Progi1984 — APPROVED** [review]

**Comment by khouloudbelguith** [comment]
Hi @dreammeup,

The shop name is well updated in the database `ps_shop`.
But this expression is not saved
Steps to reproduce the issue:
1. Go to BO > Shop Parameters > Contact > Stores page
2. Edit the shop name
3. Save => OK
4. Go to BO > Shop Parameters > Contact > Stores page again (Don't refresh the page)
5. Old value is still displayed
6. You can check BO (login page) / FO (contact information) / mails => old value is displayed

https://drive.google.com/file/d/1JM2iI4akgP6wY6cOQey8jHvCnayCWrd5/view?usp=sharing

Thank you!

**Comment by matks** [comment]
Hello, it seems this PR is unfortunately stale. 😞 QA team reported something we believe is not right and no action has been done since this moment.

I close this PR, following our [review guidelines](https://devdocs.prestashop.com/1.7/project/maintainers-guide/reviewing-pull-requests/) "Pull Requests may be closed after 30 days of inactivity". If however you are still interested in it, please ping us :) we can reopen.

Thank you for your contribution.

**Pull request closed without merging**
